cmd/object-handlers.go

```diff
@@ -976,6 +976,11 @@ func (api objectAPIHandlers) CopyObjectHandler(w http.ResponseWriter, r *http.Re
 		srcInfo.UserDefined[k] = v
 	}
 
+	// Set content-type to default value if it is not set.
+	if _, ok := srcInfo.UserDefined["Content-Type"]; !ok {
```

Comment:
This is not correct @kannappanr this has to be done differently... we have to look at the incoming headers for content-type when the "REPLACE" directive is set.